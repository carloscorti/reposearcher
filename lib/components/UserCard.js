import React from 'react';

const UserCard = ({ user }) => {
  const renderContent = (user) => {
    switch (user) {
      case null:
        return (
          <>
            <aside>
              <img src="/MVST.png" />
              <h3>MVST</h3>
              <p className="grey-text bio">React Task</p>
            </aside>
          </>
        );
      default:
        return (
          <>
            <aside>
              <img src={user.avatar_url} />
              <h3>{user.login}</h3>
              <p className="grey-text bio">{user.bio}</p>
            </aside>
          </>
        );
    }
  };

  return <>{renderContent(user)}</>;
};

export default UserCard;
