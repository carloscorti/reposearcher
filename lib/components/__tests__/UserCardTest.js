import React from 'react';
import UserCard from '../UserCard';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('UserCard', () => {
  const testProps = {
    user: {
      avatar_url: 'pic',
      login: 'userName',
      bio: 'userBio',
    },
  };

  const testPropsNullUser = {
    user: null,
  };

  it('renders correctly', () => {
    const wrapper = shallow(<UserCard user={testProps.user} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly case null repos', () => {
    const wrapper = shallow(<UserCard user={testPropsNullUser.user} />);

    expect(wrapper.find('h3').text()).toEqual('MVST');
  });
});
