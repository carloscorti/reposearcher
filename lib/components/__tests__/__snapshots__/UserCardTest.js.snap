// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renders correctly 1`] = `
<Fragment>
  <aside>
    <img
      src="pic"
    />
    <h3>
      userName
    </h3>
    <p
      className="grey-text bio"
    >
      userBio
    </p>
  </aside>
</Fragment>
`;
